File: Admin_p/Admin_p.Web/Modules/electroDB/Categories/CategoriesColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Admin_p.electroDB.Columns;

[ColumnsScript("electroDB.Categories")]
[BasedOnRow(typeof(CategoriesRow), CheckNames = true)]
public class CategoriesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int CategoryId { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string ImageUrl { get; set; }
}

File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/CategoriesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CategoriesRow } from "./CategoriesRow";

export interface CategoriesColumns {
    CategoryId: Column<CategoriesRow>;
    Name: Column<CategoriesRow>;
    Description: Column<CategoriesRow>;
    CreatedAt: Column<CategoriesRow>;
    UpdatedAt: Column<CategoriesRow>;
    ImageUrl: Column<CategoriesRow>;
}

export class CategoriesColumns extends ColumnsBase<CategoriesRow> {
    static readonly columnsKey = 'electroDB.Categories';
    static readonly Fields = fieldsProxy<CategoriesColumns>();
}